src/components/structures/RoomView.tsx

```diff
@@ -1769,7 +1926,44 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
         this.setState({ narrow });
     };
 
+    private get viewsLocalRoom(): boolean {
+        return isLocalRoom(this.state.room);
+    }
+
+    private get permalinkCreator(): RoomPermalinkCreator {
+        return this.getPermalinkCreatorForRoom(this.state.room);
```

Comment:
How does this work in the context of a local room? We don't really have a `roomId` at that stage, do we use a common mock permalink creator?

Reply:
Message composer needs that. Actually a local room shouldn't invoke it at all.  
Not sure if we need to pass a mock permalink creator.